design/vrf-announcement.md

```diff
@@ -0,0 +1,198 @@
+nodes have multiple interfaces, providing that either the CNI is able to receive
+the traffic inside the VRF or the host networking is set up in a way that the VRF
+is able to communicate with the default one for the traffic related to the
+service's IP.
+
+This is the default behaviour for a client coming from a router connected via an
+interface different from the default one.
+
+```none
+             ▲
+             │
+             │
+        ┌────┴────┐
+        │ eth0    │
+        │         │
+┌───────┴─────────┴───────────┐
+│                             │               ┌──────────────┐
+│                             │               │              │
+│                             ├────────┐      │              │
+│                             │        │      │              │
+│                             │ eth1   │◄─────┤     DCGW     │◄───────  Client
+│                             │        │      │              │
+│                             ├────────┘      │              │
+│                             │               │              │
+│                      Node   │               └──────────────┘
+│                             │
+└─────────────────────────────┘
+```
+
+And this is how it would look like with VRF:
+
+```none
+
+
+
+               ┌─────────┐
+               │ eth0    │
+               │         │
+       ┌───────┴─────────┴───────────┐
+       │                             │               ┌──────────────┐
+       │                      ┌──────┤               │              │
+       │                      │      ├────────┐      │              │
+       │                      │      │        │      │              │
+       │                      │ VRF  │ eth1   │◄─────┤     DCGW     │◄───────  Client
+       │                      │      │        │      │              │
+       │                  ────┼────► ├────────┘      │              │
+       │                      └──────┤               │              │
+       │                      Node   │               └──────────────┘
+       │                             │
+       └─────────────────────────────┘
+
+```
+
+Other advantages of having VRFs on the node are:
+
+* being able to have more complex routing tables (i.e. multiple DCGWs for HA)
+without impacting the node networking
+* ease the troubleshooting of the traffic directed to the service
+* making it possible to have multiple clients (coming from different interfaces)
+sharing the same IP
+
+VRF for BGP is supported by FRR [out of the box](https://docs.frrouting.org/en/latest/bgp.html#clicmd-router-bgp-ASN-vrf-VRFNAME).
+
+### Goals
+
+* Extablish a BGP session with a BGP peer using an interface with VRF as master
+* Announce a LoadBalancer service to a BGP peer peered via an interface belonging
+to a VRF
+* Eventually, provide a guideline and an example on how to setup the VRF and the
+host routing, knowing that the routing logic might be CNI dependant
+
+### Non-Goals
+
+* Creating the VRF
```

Comment:
what about reacting to updates to a vrf (renames/deletions/?)? I assume we do not care about this, but think it is worth mentioning here so everything is clear (what MetalLB cares about regarding VRFs)
also, if a speaker's host does not have the vrf configured what's the behavior?